src/fr/scanmanga/src/eu/kanade/tachiyomi/extension/fr/scanmanga/ScanManga.kt

```diff
@@ -78,119 +86,87 @@ class ScanManga : ParsedHttpSource() {
-            n--
-        }
-        return result.toString()
-    }
-
     override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
-        val searchHeaders = headersBuilder()
-            .add("Referer", "$baseUrl/scanlation/liste_series.html")
-            .add("x-requested-with", "XMLHttpRequest")
+        val url = desktopBaseUrl.toHttpUrlOrNull()!!.newBuilder()
+            .addPathSegments("scanlation/liste_series.html")
+            .addQueryParameter("q", query)
             .build()
-
-        return GET("$baseUrl/scanlation/scan.data.json", searchHeaders)
+            .toString()
+
+        return GET(
+            url,
+            headersBuilder()
+                .add("User-Agent", "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/115.0 Safari/537.36")
+                .add("Accept-Language", "fr-FR,fr;q=0.9,en-US;q=0.8,en;q=0.7")
+                .add("Referer", desktopBaseUrl)
+                .build(),
+        )
     }
 
-    override fun fetchSearchManga(page: Int, query: String, filters: FilterList): Observable<MangasPage> {
-        return client.newCall(searchMangaRequest(page, query, filters))
-            .asObservableSuccess()
-            .map { response ->
-                searchMangaParse(response, query)
-            }
-    }
+    override fun searchMangaSelector() = "div.texte_manga.book_close, div.texte_manga.book_stop"
+
+    override fun searchMangaFromElement(element: Element): SManga {
+        val manga = SManga.create()
+        val link = element.selectFirst("a.texte_manga")!!
 
-    private fun searchMangaParse(response: Response, query: String): MangasPage {
-        return MangasPage(parseMangaFromJson(response).mangas.filter { it.title.contains(query, ignoreCase = true) }, false)
+        val path = link.attr("href")
+        // Construction correcte de l'URL vers la version mobile
+        manga.setUrlWithoutDomain(path)
+        manga.url = if (path.startsWith("/")) "https://m.scan-manga.com$path" else path
+
+        manga.title = link.text().trim()
+        manga.thumbnail_url = null // Pas d'image dispo ici
+        return manga
     }
 
-    private fun parseMangaFromJson(response: Response): MangasPage {
-        val jsonRaw = response.body.string()
+    override fun searchMangaNextPageSelector(): String? = null
 
-        if (jsonRaw.isEmpty()) {
-            return MangasPage(emptyList(), hasNextPage = false)
+    // Details
+    override fun mangaDetailsParse(document: Document): SManga {
+        val manga = SManga.create()
+
+        manga.title = document.select("h1.main_title[itemprop=name]").text()
+        manga.author = document.select("div[itemprop=author]").text()
+        manga.description = document.select("div.titres_desc[itemprop=description]").text()
+        manga.genre = document.select("div.titres_souspart span[itemprop=genre]").joinToString(", ") { it.text() }
```

Comment:
(nit) `, ` is already the default.
```suggestion
        manga.genre = document.select("div.titres_souspart span[itemprop=genre]").joinToString { it.text() }
```